### PR TITLE
Improvement of Exception info in File manipulation.

### PR DESCRIPTION
**Pull request by dgedgedge**
Improvement of Exception through.
Not tested. I do not have the mean.

I did this to help debug an issue with nextcloud/Music application.
It hanged while scanning files.

Signed-off-by: Denis Germa <denis.germa@free.fr>

### CODE REVIEW

File: lib/private/Files/Node/File.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function getContent() {
 			 */
 			return $this->view->file_get_contents($this->path);
 		} else {
-			throw new NotPermittedException();
+			throw new NotPermittedException('Permission issue unable to get content of file ' . $this->path);
```
Comment (kesselb) [inline lib/private/Files/Node/File.php:58]: > Permission issue unable to get content of file

I don't think so. We don't have permission to read the file.

File: lib/private/Files/Node/File.php [hunk 2]
```diff
@@ -74,7 +74,7 @@ public function putContent($data) {
 			$this->fileInfo = null;
 			$this->sendHooks(array('postWrite'));
 		} else {
-			throw new NotPermittedException();
+			throw new NotPermittedException('Permission issue unable to get content of file ' . $this->path);
```
Comment (kesselb) [inline lib/private/Files/Node/File.php:77]: We don't have permission to update a file here. Unable to get content of file is not correct.

File: lib/private/Files/Node/File.php [hunk 3]
```diff
@@ -115,7 +115,7 @@ public function fopen($mode) {
 			$this->sendHooks($postHooks);
 			return $result;
 		} else {
-			throw new NotPermittedException();
+			throw new NotPermittedException('Permission issue Could not open (Incorrect mode ' . $mode . ') ' . $this->path);
```
Comment (kesselb) [inline lib/private/Files/Node/File.php:118]: The mode is not incorrect. We don't have the required permissions to do that operation. Example: w+ requires us to have read and update permission for a file.

**Review by kesselb — COMMENTED** [review]
Thanks :+1: It's a good idea to add more details to those exceptions. But the current state adds more confusion from my pov.

**Comment by MorrisJobke** [comment]
Base repository was deleted. And there was no feedback since a while. Thus I will close this PR for now. It can be picked up later of course.

**Pull request closed without merging**